packages/monaco/src/browser/monaco-quick-open-service.ts

```diff
@@ -19,34 +19,263 @@ import { MessageType } from '@theia/core/lib/common/message-service-protocol';
 import {
     QuickOpenService, QuickOpenModel, QuickOpenOptions, QuickOpenItem, QuickOpenGroupItem,
     QuickOpenMode, KeySequence, QuickOpenActionProvider, QuickOpenAction, ResolvedKeybinding,
-    KeyCode, Key, KeybindingRegistry
+    KeyCode, Key, KeybindingRegistry, TitleButton, FOLDER_ICON, FILE_ICON
 } from '@theia/core/lib/browser';
 import { KEY_CODE_MAP } from './monaco-keycode-map';
 import { ContextKey } from '@theia/core/lib/browser/context-key-service';
 import { MonacoContextKeyService } from './monaco-context-key-service';
+import { Emitter, Event } from '@theia/core/lib/common/event';
+import { BuiltinThemeProvider, ThemeService } from '@theia/core/lib/browser/theming';
+import URI from '@theia/core/lib/common/uri';
+import { DisposableCollection } from '@theia/core';
 
 export interface MonacoQuickOpenControllerOpts extends monaco.quickOpen.IQuickOpenControllerOpts {
+    busy?: boolean
+    enabled?: boolean;
+    step?: number | undefined
+    title?: string | undefined
+    totalSteps?: number | undefined
+    buttons?: ReadonlyArray<TitleButton>
+
     readonly prefix?: string;
     readonly password?: boolean;
     readonly ignoreFocusOut?: boolean;
     onType?(lookFor: string, acceptor: (model: monaco.quickOpen.QuickOpenModel) => void): void;
     onClose?(canceled: boolean): void;
 }
 
+class MonacoTitleBar {
+
+    private readonly onDidTriggerButtonEmitter: Emitter<TitleButton>;
+    private _isAttached: boolean;
```

Comment:
please study coding guidlines, `_` is allowed only in special cases

please align styles where applicable